packages/story-editor/src/components/mediaRecording/useTrim.js

```diff
@@ -17,30 +17,46 @@
 /**
  * External dependencies
  */
+import { formatMsToHMS } from '@googleforcreators/media';
 import { useState, useCallback } from '@googleforcreators/react';
 
-function useTrim(setDuration) {
+/**
+ * Internal dependencies
+ */
+import useFFmpeg from '../../app/media/utils/useFFmpeg';
+
+function useTrim({ setDuration, onTrimmed, file }) {
   const [trimData, setTrimData] = useState({ start: 0, end: null });
   const [isTrimming, setIsTrimming] = useState(false);
+  const [isProcessingTrim, setIsProcessingTrim] = useState(false);
+  const { trimVideo } = useFFmpeg();
   const onTrim = useCallback(
-    (newTrimData) => {
+    async (newTrimData) => {
+      setIsTrimming(false);
       if (newTrimData) {
         setTrimData(newTrimData);
-        const millis = newTrimData.end - newTrimData.start;
-        const seconds = Math.ceil(millis / 1000);
-        setDuration(seconds);
+        setIsProcessingTrim(true);
```

Comment:
What's the difference between `processingTrim` and `isTrimming`? Does `processingTrim` include waiting until the playback has finished and `isTrimming` doesn't? Would be great to have a comment for readability to differentiate these two.